test/Microsoft.Windows.CsWin32.Tests/GeneratorTests.cs

```diff
@@ -377,6 +371,78 @@ internal enum FILE_CREATE_FLAGS
         this.AssertNoDiagnostics();
     }
 
+    // Slow span can't safely create a span from a ref when the struct is on the heap.
+    [Fact]
+    public void CodeGenerationForFixedLengthInlineArrayWithSlowSpan()
+    {
+        const string expected = @"
+    internal partial struct MainAVIHeader
+    {
+        internal uint dwMicroSecPerFrame;
+        internal uint dwMaxBytesPerSec;
+        internal uint dwPaddingGranularity;
+        internal uint dwFlags;
+        internal uint dwTotalFrames;
+        internal uint dwInitialFrames;
+        internal uint dwStreams;
+        internal uint dwSuggestedBufferSize;
+        internal uint dwWidth;
+        internal uint dwHeight;
+        internal __dwReserved_4 dwReserved;
+        [StructLayout(LayoutKind.Sequential, Pack = 1)]
+        internal struct __dwReserved_4
+        {
+            internal uint _1, _2, _3, _4;
+            internal ref uint this[int index]
+            {
+                [MethodImpl(MethodImplOptions.AggressiveInlining)]
+                get
+                {
+                    unsafe
+                    {
+                        fixed (uint *p = &_1)
+                            return ref p[index];
+                    }
+                }
+            }
+        }
+    }
+";
+
+        this.AssertGeneratedType("MainAVIHeader", expected);
+    }
+
+    [Fact]
+    public void CodeGenerationForFixedLengthInlineArrayWithFastSpan()
+    {
+        const string expected = @"
+    internal partial struct MainAVIHeader
+    {
+        internal uint dwMicroSecPerFrame;
+        internal uint dwMaxBytesPerSec;
+        internal uint dwPaddingGranularity;
+        internal uint dwFlags;
+        internal uint dwTotalFrames;
+        internal uint dwInitialFrames;
+        internal uint dwStreams;
+        internal uint dwSuggestedBufferSize;
+        internal uint dwWidth;
+        internal uint dwHeight;
+        internal __dwReserved_4 dwReserved;
+        [StructLayout(LayoutKind.Sequential, Pack = 1)]
+        internal struct __dwReserved_4
+        {
+            internal uint _1, _2, _3, _4;
+            internal ref uint this[int index] => ref AsSpan()[index];
+            internal Span<uint> AsSpan() => MemoryMarshal.CreateSpan(ref _1, 4);
+        }
+    }
+";
```

Comment:
I'm not a fan of character-level equality checks as I expect they will cost too much to maintain. Rather, specific API traits can be asserted as other tests in this class do, and for functional testing, see `BasicTests.cs`. But we can keep this if you like, at least until it requires too many updates. :) It certainly helps in a PR review to see what you expect.